src/tests/ftest/util/server_utils.py

```diff
@@ -50,6 +57,439 @@ class ServerFailed(Exception):
+                Defaults to 30.
+        """
+        super(ServerManager, self).__init__("/run/server_manager/*", "", "")
+
+        self.daosbinpath = daosbinpath
+        self._hosts = None
+
+        # Setup orterun command defaults
+        self.runner = Orterun(
+            DaosServer(self.daosbinpath), runnerpath, True)
+
+        # Setup server command defaults
+        self.runner.job.timeout = timeout
+        self.runner.job.action.value = "start"
+        self.runner.job.set_action_command()
+
+        # Parameters that user can specify in the test yaml to modify behavior.
+        self.debug = BasicParameter(None, True)       # ServerCommand param
+        self.attach = BasicParameter(None, attach)    # ServerCommand param
+        self.insecure = BasicParameter(None, True)    # ServerCommand param
+        self.sudo = BasicParameter(None, False)       # ServerCommand param
+        self.report_uri = BasicParameter(None)             # Orterun param
+        self.enable_recovery = BasicParameter(None, True)  # Orterun param
+        self.export = BasicParameter(None)                 # Orterun param
+
+    @property
+    def hosts(self):
+        """Hosts attribute getter."""
+        return self._hosts
+
+    @hosts.setter
+    def hosts(self, value):
+        """Hosts attribute setter
+
+        Args:
+            value (tuple): (list of hosts, workdir, slots)
+        """
+        self._hosts, workdir, slots = value
+        self.runner.processes.value = len(self._hosts)
+        self.runner.hostfile.value = write_host_file(
+            self._hosts, workdir, slots)
+        self.runner.job.server_cnt = len(self._hosts)
+
+    def get_params(self, test):
+        """Get values from the yaml file and assign them respectively
+            to the server command and the orterun command.
+
+        Args:
+            test (Test): avocado Test object
+        """
+        server_params = ["debug", "sudo"]
+        server_start_params = ["attach", "insecure"]
+        runner_params = ["enable_recovery", "export", "report_uri"]
+        super(ServerManager, self).get_params(test)
+        self.runner.job.yaml_params.get_params(test)
+        self.runner.get_params(test)
+        for name in self.get_param_names():
+            if name in server_params:
+                if name == "sudo":
+                    setattr(self.runner.job, name, getattr(self, name).value)
+                else:
+                    getattr(
+                        self.runner.job, name).value = getattr(self, name).value
+            if name in server_start_params:
+                getattr(self.runner.job.action_command, name).value = \
+                    getattr(self, name).value
+            if name in runner_params:
+                getattr(self.runner, name).value = getattr(self, name).value
+
+    def run(self):
+        """Execute the runner subprocess."""
+        return self.runner.run()
```

Comment:
To retain the same logging as the run_server() function it essentially replaces, I'd suggest inserting the following before this line:
`self.log.info("Start CMD>>> %s", str(self.runner))`